### ISSUE TITLE
2.6.0 cannot compile rotating-workforce instances

### ISSUE THREAD

**Issue by informarte** [MiniZinc/libminizinc#552]
```
minizinc rotating-workforce.mzn Example103.dzn

Error: evaluation error: indexes don't match size of generated array
/home/informarte/Workspaces/yuck/resources/mzn/tests/minizinc-benchmarks/rotating-workforce/rotating-workforce.mzn:238.1-39
  in variable declaration for 'sort_min_slack3'
  in call 'int2float'
/home/informarte/Software/lib/MiniZincIDE-2.6.0-bundle-linux-x86_64/share/minizinc/std/stdlib/stdlib_coercion.mzn:128.3-130.57
  in let expression
/home/informarte/Software/lib/MiniZincIDE-2.6.0-bundle-linux-x86_64/share/minizinc/std/stdlib/stdlib_coercion.mzn:130.8-57
  in call 'arrayXd'
/home/informarte/Software/lib/MiniZincIDE-2.6.0-bundle-linux-x86_64/share/minizinc/std/stdlib/stdlib_array.mzn:113.3-117.7
  in if-then-else expression
/home/informarte/Software/lib/MiniZincIDE-2.6.0-bundle-linux-x86_64/share/minizinc/std/stdlib/stdlib_array.mzn:116.5-39
  in array comprehension expression
```

**Comment by alexeytref** [comment]
I think the problem is the same, but I'll describe it just in case. The code `array [1..5] of float: x = [17, 64, 14, 32, 15];` fails with an error:
_Playground:1.1-33
  in variable declaration for 'DemandLoad'
  in call 'int2float'
stdlib_coercion:128-3.130-57
  in let expression
stdlib_coercion:130.8-57
  in call 'arrayXd'
stdlib_array:113-3.117-7
  in if-then-else expression
stdlib_array:116.5-39
  in array comprehension expression
MiniZinc: evaluation error: indexes don't match size of generated array
Process finished with non-zero exit code 1.
Finished in 196msec._
about every tenth run.

**Comment by cyderize** [comment]
@alexeytref 
Yes, that's the same problem and will be fixed in 2.6.1.
